### ISSUE TITLE
SELECT command denied on cleanup

### ISSUE THREAD

**Issue by pacija** [ampache/ampache#1659]
# Environment

 * Ampache version: 3.8.4-release 
 * Web server + version: apache24-2.4.29
 * Server operating system: FreeBSD 11.1-RELEASE-p6

# Settings

```
config_version = 34
web_path = ""
database_hostname = "localhost"
database_port = ""
database_name = "database_name"
database_username = "database_username"
database_password = "database_password"
secret_key = "secret_key"
session_length = 3600
stream_length = 7200
remember_length = 604800
session_name = ampache
session_cookielife = 0
session_cookiesecure = 1
auth_methods = "mysql"
access_control	= "true"
require_session = "true"
require_localnet_session = "true"
getid3_tag_order = "id3v2"
metadata_order = "getID3"
metadata_order_video = "getID3"
deferred_ext_metadata = "true"
additional_genre_delimiters = "[/]{2}|[/|\\\\|\|,|;]"
catalog_file_pattern = "mp3|mpc|m4p|m4a|aac|ogg|oga|wav|aif|aiff|rm|wma|asf|flac|opus|spx|ra|ape|shn|wv"
catalog_video_pattern = "avi|mpg|mpeg|flv|m4v|mp4|webm|mkv|wmv|ogv|mov|divx|m2ts"
catalog_playlist_pattern = "m3u|m3u8|pls|asx|xspf"
catalog_prefix_pattern = "The|An|A|Die|Das|Ein|Eine|Les|Le|La"
track_user_ip = "true"
no_symlinks	= "true"
use_auth = "true"
default_auth_level = "guest"
ratings = "true"
userflags = "true"
directplay = "true"
sociable = "true"
licensing = "false"
memory_cache = "true"
memory_limit = 64
art_order = "db,tags"
lastfm_api_key = "d5df942424c71b754e54ce1832505ae2"
lastfm_api_secret = ""
wanted = "true"
wanted_types = "album,official"
channel = "true"
live_stream = "true"
refresh_limit = "60"
show_footer_statistics = "true"
debug = "true"
debug_level = 5
log_path = "/var/log/ampache"
log_filename = "%name.%Y%m%d.log"
site_charset = "UTF-8"
allow_public_registration = "false"
cookie_disclaimer = "false"
registration_display_fields = "fullname,website"
registration_mandatory_fields = "fullname"
transcode_player_customize = "true"
transcode_input = "-i %FILE%"
encode_args_mp3 = "-vn -b:a %BITRATE%K -c:a libmp3lame -f mp3 pipe:1"
encode_args_ogg = "-vn -b:a %BITRATE%K -c:a libvorbis -f ogg pipe:1"
encode_args_opus = "-vn -b:a %BITRATE%K -c:a libopus -compression_level 10 -vsync 2 -f ogg pipe:1"
encode_args_m4a = "-vn -b:a %BITRATE%K -c:a libfdk_aac -f adts pipe:1"
encode_args_wav = "-vn -b:a %BITRATE%K -c:a pcm_s16le -f wav pipe:1"
encode_args_flv = "-b:a %BITRATE%K -ar 44100 -ac 2 -v 0 -f flv -c:v libx264 -preset superfast -threads 0 pipe:1"
encode_args_webm = "-q %QUALITY% -f webm -c:v libvpx -maxrate %MAXBITRATE%k -preset superfast -threads 0 pipe:1"
encode_args_ts = "-q %QUALITY% -s %RESOLUTION% -f mpegts -c:v libx264 -c:a libmp3lame -maxrate %MAXBITRATE%k -preset superfast -threads 0 pipe:1"
encode_get_image = "-ss %TIME% -f image2 -vframes 1 pipe:1"
encode_srt = "-vf \"subtitles='%SRTFILE%'\""
encode_ss_frame = "-ss %TIME%"
encode_ss_duration = "-t %DURATION%"
common_abbr = "divx,xvid,dvdrip,hdtv,lol,axxo,repack,xor,pdtv,real,vtv,caph,2hd,proper,fqm,uncut,topaz,tvt,notv,fpn,fov,orenji,0tv,omicron,dsr,ws,sys,crimson,wat,hiqt,internal,brrip,boheme,vost,vostfr,fastsub,addiction,x264,LOL,720p,1080p,YIFY,evolve,fihtv,first,bokutox,bluray,tvboom,info"
```

# Description

Cleaning catalogues apparently does nothing, ends up with error log message.

# Logs
```
2018-01-16 11:20:02 [admin] (Dba) -> Error: ["42000",1142,"SELECT command denied to user 'database_user'@'localhost' for table 'song'"] 
2018-01-16 11:20:02 [admin] (Dba) -> Error: ["42000",1142,"SELECT command denied to user 'database_user'@'localhost' for table 'song'"] 
2018-01-16 11:20:02 [admin] (Dba) -> Error: ["42000",1142,"SELECT command denied to user 'database_user'@'localhost' for table 'song'"] 
2018-01-16 11:20:02 [admin] (Dba) -> Error: ["42000",1142,"SELECT command denied to user 'database_user'@'localhost' for table 'song'"] 
```

**Comment by wagnered** [comment]
@pacija 

`database_username` in the config file above, must be set to a user who was granted  access to the database (usually root) during the install process.

If you created an alternate db user, when Ampache was installed, perhaps the process of giving the new db user permissions, was interrupted and didn't get completed. Or the `database_name` or `database_username` in the config file above has been changed.

**Comment by pacija** [comment]
@wagnered 

I have replaced real `database_name`, `database_username` and `database_password` in config and log above so their values are the same as variables.

Empty database `database_name` was created in phpmyadmin beforehand. `database_username` was granted `ALL PRIVILEGES` on `database_name`.

My ampache setup works, I can create catalogs, listen to music etc. If I delete all catalogs and hit "clean all", artist and album tables in database get emptied. So I would conclude above log message doesn't have much to do with mysql credentials, as you suggest. I don't see any problem in functionality because of this message. I reported it here because it looks odd to me.
